src/ui/common/src/components/workflows/ReactFlowCanvas.tsx

```diff
@@ -44,11 +52,148 @@ const ReactFlowCanvas: React.FC<ReactFlowCanvasProps> = ({
     setTimeout(fitView, 200);
   }, [openSideSheetState]);
 
+  const checkOpNodes = [];
+  const boolArtifactNodes = [];
+
+  const metricOpNodes = [];
+  const metricArtifactNodes = [];
+
+  // first find all check operators.
+  if (dagPositionState.result) {
+    const nodes = dagPositionState.result.nodes;
+    nodes.forEach((node) => {
+      if (node.type === 'checkOp') {
+        checkOpNodes.push(node);
+      } else if (node.type === 'boolArtifact') {
+        boolArtifactNodes.push(node);
+      } else if (node.type === 'metricOp') {
+        metricOpNodes.push(node);
+      } else if (node.type === 'numericArtifact') {
+        metricArtifactNodes.push(node);
+      }
+    });
+
+    //sort checkOpNodes and boolArtifactNodes by value of label
+    //operator nodes have just a name
+    //artifact nodes have same name + ' artifact' at the end.
+    checkOpNodes.sort((a, b) => a.data.label.localeCompare(b.data.label));
```

Comment:
could take the sorting function here and abstract it out so we can reuse with the other node types.

e.g.
```
const alphabeticalSort = (a, b) => a.data.label.localeCompare(b.data.label);
```

Bonus points for using the correct types for a and b (whatever the node type is)